File: litsea-ruby/examples/segment.rb
# frozen_string_literal: true

# Segment a sentence, with and without POS tags.
#
# Usage:
#   bundle exec ruby -Ilib examples/segment.rb ../models/japanese.model "これはテストです。"

require 'litsea'

model_path, text = ARGV
if model_path.nil? || text.nil?
  warn 'usage: ruby examples/segment.rb <model> <text>'
  exit 2
end

# The model file identifies its own kind, so nothing here declares whether
# this is a POS model - has_pos? reports what was loaded.
segmenter = Litsea::Segmenter.open(:japanese, model_path)
puts "model: #{model_path} (has_pos?=#{segmenter.has_pos?})"
puts "tokens: #{segmenter.segment(text).join(' ')}"

return unless segmenter.has_pos?

puts 'tagged:'
segmenter.segment_with_pos(text).each do |token|
  puts format("  %<surface>s\t%<pos>s\t[%<start>d:%<end>d]",
              surface: token.surface, pos: token.pos, start: token.start, end: token.end)
end

File: litsea-nodejs/examples/segment.mjs
/**
 * Segment a sentence, with and without POS tags.
 *
 * Usage:
 *   node examples/segment.mjs ../models/japanese.model "これはテストです。"
 *   node examples/segment.mjs ../models/japanese_pos.model "これはテストです。"
 */

import { Segmenter } from '../index.js'

const [, , modelPath, text] = process.argv

if (!modelPath || !text) {
  console.error('usage: node examples/segment.mjs <model> <text>')
  process.exit(2)
}

// The model file identifies its own kind, so nothing here declares whether
// this is a POS model - `hasPos` reports what was loaded.
const segmenter = Segmenter.open('japanese', modelPath)
console.log(`model: ${modelPath} (hasPos=${segmenter.hasPos})`)
console.log('tokens:', segmenter.segment(text).join(' '))

if (segmenter.hasPos) {
  console.log('tagged:')
  for (const token of segmenter.segmentWithPos(text)) {
    console.log(`  ${token.surface}\t${token.pos}\t[${token.start}:${token.end}]`)
  }
}
